### PR TITLE
Remove common adapter

### PR DESCRIPTION
**Pull request by gerben-stavenga**
### Description
1) Remove adapter_common which defines a trait that isn't productively used
2) Remove an arc for thread_pool which should be a simple reference
3) Merge AptosVM and AptosVMImpl which simplifies code

### CODE REVIEW

**Comment by davidiw** [comment]
This is pretty cool. I do wonder if we can at least some amount of logic separation so we don't end up with a mindlessly long single file.

File: aptos-move/aptos-vm/src/aptos_vm.rs [hunk 1]
```diff
@@ -60,39 +65,40 @@ use move_core_types::{
...
 
-pub static RAYON_EXEC_POOL: Lazy<Arc<rayon::ThreadPool>> = Lazy::new(|| {
-    Arc::new(
-        rayon::ThreadPoolBuilder::new()
-            .num_threads(num_cpus::get())
-            .thread_name(|index| format!("par_exec_{}", index))
-            .build()
-            .unwrap(),
-    )
+static RAYON_EXEC_POOL: Lazy<rayon::ThreadPool> = Lazy::new(|| {
```
Comment (gelash) [inline aptos-move/aptos-vm/src/aptos_vm.rs:96]: The arc change looks great, but I see a different potential issue, let's discuss here but doesn't need to be addressed here.

@sitalkedia @zekun000 @danielxiangzl 
Having the RAYON_EXEC_POOL here seems dangerous to me (sorry that I couldn't review the original PR that moved it in time) - the problem is, the way our current Block-STM implementation uses it, with long-running workers that may sleep, it's almost guaranteed that we can and will deadlock if someone uses the threadpool concurrently with block execution. And here we receive all the callbacks from the Move-VM, so it's super easy to make that mistake.

I think we can move it to block_executor/mod.rs maybe, but not here..

File: aptos-move/aptos-vm/src/aptos_vm.rs [hunk 2]
```diff
@@ -114,9 +140,99 @@ macro_rules! unwrap_or_discard {
...
+        let mut vm = Self {
+            move_vm: inner,
+            gas_feature_version,
+            gas_params,
+            storage_gas_params,
+            version: None,
+            transaction_validation: None,
+            features,
+        };
+        vm.version = Version::fetch_config(&storage);
```
Comment (gelash) [inline aptos-move/aptos-vm/src/aptos_vm.rs:233]: why are we assigning these members here right after creation? wouldn't Self { ... vesion: Version::fetch_config, transaction_validation: Self::get_transaction_validation(..)..} work?

or we could assign to version and transaction_validation variables earlier (like inner)

File: aptos-move/aptos-vm/src/aptos_vm.rs [hunk 3]
```diff
@@ -199,8 +315,11 @@ impl AptosVM {
         }
     }
 
-    pub fn internals(&self) -> AptosVMInternals {
-        AptosVMInternals::new(&self.0)
+    pub fn version(&self) -> Result<Version, VMStatus> {
```
Comment (gelash) [inline aptos-move/aptos-vm/src/aptos_vm.rs:318]: This was/is very confusing: version only seems to be used for tests, and with unwrap, yet, alert has a quite scary message ("VM Startup failed"). Where do we even use this version, does it need to be Option, or should the error on None just be handled by the new() method?

**Comment by gelash** [comment]
> This is pretty cool. I do wonder if we can at least some amount of logic separation so we don't end up with a mindlessly long single file.

maybe we can isolate some of the gas logic that was previously in the deleted files as utility methods. Those have TODOs and are likely to evolve anyway, so some isolation there could help.

File: aptos-move/aptos-vm/src/aptos_vm.rs [hunk 4]
```diff
@@ -114,9 +140,99 @@ macro_rules! unwrap_or_discard {
...
 }
 
+/// Public interface for the VM.
 impl AptosVM {
+    #[allow(clippy::new_without_default)]
     pub fn new(state: &impl StateView) -> Self {
-        Self(AptosVMImpl::new(state))
+        let storage = StorageAdapter::new(state);
+
+        // Get the gas parameters
```
Comment (gelash) [inline aptos-move/aptos-vm/src/aptos_vm.rs:149]: maybe some of this gas things can be extracted out to utils.rs (or gas_utils.rs if there are more things)

File: aptos-move/aptos-vm/src/aptos_vm.rs [hunk 5]
```diff
@@ -1535,137 +1776,441 @@ impl VMValidator for AptosVM {
     }
 }
 
-impl VMAdapter for AptosVM {
-    fn new_session<'r>(
+pub(crate) fn gas_config(storage: &impl MoveResolverExt) -> (Option<AptosGasParameters>, u64) {
```
Comment (gelash) [inline aptos-move/aptos-vm/src/aptos_vm.rs:1779]: more candidates to be extracted to gas_utils

File: aptos-move/aptos-vm/src/aptos_vm.rs [hunk 6]
```diff
@@ -1535,137 +1776,441 @@ impl VMValidator for AptosVM {
...
+        self.move_vm.mark_loader_cache_as_invalid();
+    }
+
+    fn transaction_validation(&self) -> &TransactionValidation {
+        self.transaction_validation
+            .as_ref()
+            .unwrap_or(&APTOS_TRANSACTION_VALIDATION)
+    }
+
+    // TODO: Move this to an on-chain config once those are a part of the core framework
```
Comment (gelash) [inline aptos-move/aptos-vm/src/aptos_vm.rs:1811]: This TODO seems resolved?

**Review by gelash — APPROVED** [review]

**Comment by davidiw** [comment]
let's resurrect this in the near future...

**Comment by gelash** [comment]
I will commandeer this.

**Comment by gelash** [comment]
but @davidiw lmk if I should close and open a new one

**Comment by gelash** [comment]
will open a new PR.

**Pull request closed without merging**